release/src/module_dependencies/main.py

```diff
@@ -2,33 +2,59 @@
 import json
 import base64
 import networkx as nx
+import sys
+from retry import retry
 
 def main():
+    print('Running main.py')
     moduleNameList = sortModuleNameList()
+    print('Fetched module name list')
     moduleDetailsJSON = initializeModuleDetails(moduleNameList)
+    print('Initialized module details and fetched latest module versions')
     moduleDetailsJSON = getImmediateDependents(moduleNameList, moduleDetailsJSON)
+    print('Fetched immediate dependents of each module')
     moduleDetailsJSON = calculateLevels(moduleNameList, moduleDetailsJSON)
+    print('Generated module dependency graph and updated module levels')
     updateJSONFile(moduleDetailsJSON)
+    print('Updated module details successfully')
 
 # Sorts the ballerina standard library module list in ascending order
 def sortModuleNameList():
-    with open('./release/resources/module_list.json') as f:
-          nameList = json.load(f)
+    try:
+        with open('./release/resources/module_list.json') as f:
+            nameList = json.load(f)
+    except:
+        print('Failed to read module_list.json')
+        sys.exit()
 
     nameList['modules'].sort()
     
-    with open('./release/resources/module_list.json', 'w') as jsonFile:
-        jsonFile.seek(0) 
-        json.dump(nameList, jsonFile, indent=4)
-        jsonFile.truncate()
+    try:
+        with open('./release/resources/module_list.json', 'w') as jsonFile:
+            jsonFile.seek(0) 
+            json.dump(nameList, jsonFile, indent=4)
+            jsonFile.truncate()
+    except:
+        print('Failed to write to file module_list.json')
+        sys.exit()
         
     return nameList['modules'] 
 
+# Returns the file in the given url
+# Retry decorator will retry the function 3 times, doubling the backoff delay if URLError is raised 
+@retry(urllib.error.URLError, tries=3, delay=2, backoff=2)
```

Comment:
What is the unit of this delay? 

Shall we define proper constants for the `tries`, `delay`, and the `backoff` values and use them here?